File: Planarian/Planarian/Modules/Caves/Models/AddCaveVm.cs
// using System.Collections;
// using System.ComponentModel.DataAnnotations;
// using Planarian.Model.Database.Entities.RidgeWalker.ViewModels;
// using Planarian.Model.Shared;
// using Planarian.Modules.Files.Controllers;
//
// namespace Planarian.Modules.Caves.Models;
//
// public class AddCaveVm : AddCave
// {
//     public AddCaveVm(AddCave cave)
//     {
//         Id = cave.Id;
//         Name = cave.Name;
//         AlternateNames = cave.AlternateNames;
//
//         CountyId = cave.CountyId;
//         StateId = cave.StateId;
//         LengthFeet = cave.LengthFeet;
//         DepthFeet = cave.DepthFeet;
//         MaxPitDepthFeet = cave.MaxPitDepthFeet;
//         NumberOfPits = cave.NumberOfPits;
//
//         Narrative = cave.Narrative;
//         ReportedOn = cave.ReportedOn;
//         ReportedByName = cave.ReportedByName;
//
//         Entrances = cave.Entrances;
//
//         Files = cave.Files;
//
//         GeologyTagIds = cave.GeologyTagIds;
//
//         ReportedByNameTagIds = cave.ReportedByNameTagIds;
//
//         BiologyTagIds = cave.BiologyTagIds;
//
//         ArcheologyTagIds = cave.ArcheologyTagIds;
//
//         CartographerNameTagIds = cave.CartographerNameTagIds;
//
//         MapStatusTagIds = cave.MapStatusTagIds;
//
//         GeologicAgeTagIds = cave.GeologicAgeTagIds;
//
//         PhysiographicProvinceTagIds = cave.PhysiographicProvinceTagIds;
//
//         OtherTagIds = cave.OtherTagIds;
//     }
//
//     public static AddCaveVm FromAddCave(AddCave cave) => new AddCaveVm(cave);
// }

File: Planarian.Web/src/Modules/Caves/Models/AddCaveVm.ts
import { Dayjs } from "dayjs";
import { GeoJsonUploadVm } from "./GeoJsonUploadVm";
import { EditFileMetadataVm } from "../../Files/Models/EditFileMetadataVm";
import { AddEntranceVm } from "./AddEntranceVm";

export interface AddCaveVm {
  id?: string;
  expectedRevisionId?: string | null;
  name: string;
  alternateNames: string[];
  countyId: string | null;
  stateId: string;
  countyDisplayId?: string | null;
  countyNumber?: number | null;
  isCountyNumberManuallySet: boolean;
  useFirstAvailableCountyNumber: boolean;
  lengthFeet: number | null;
  depthFeet: number | null;
  maxPitDepthFeet: number | null;
  numberOfPits: number | null;
  narrative: string | null;
  reportedOn: Dayjs | null;
  entrances: AddEntranceVm[];
  geologyTagIds: string[];
  files?: EditFileMetadataVm[];
  linePlots?: GeoJsonUploadVm[];
  reportedByNameTagIds: string[];
  biologyTagIds: string[];
  archeologyTagIds: string[];
  cartographerNameTagIds: string[];
  mapStatusTagIds: string[];
  geologicAgeTagIds: string[];
  physiographicProvinceTagIds: string[];
  otherTagIds: string[];
}
